### PR TITLE
Codec interface tests

### PR DESCRIPTION
**Pull request by ilija42**

### CODE REVIEW

File: pkg/solana/chainreader/chain_reader.go [hunk 1]
```diff
@@ -223,12 +223,12 @@ func (s *SolanaChainReaderService) CreateContractType(readIdentifier string, for
...
 	for namespace, methods := range namespaces {
 		for methodName, method := range methods.Methods {
-			var idl codec.IDL
+			var idl solanacodec.IDL
 			if err := json.Unmarshal([]byte(method.AnchorIDL), &idl); err != nil {
 				return err
 			}
 
-			idlCodec, err := codec.NewIDLAccountCodec(idl, config.BuilderForEncoding(method.Encoding))
+			idlCodec, err := solanacodec.NewIDLAccountCodec(idl, config.BuilderForEncoding(method.Encoding))
```
Comment (ilija42) [inline pkg/solana/chainreader/chain_reader.go]: Will replace these in another PR to initialise Codec in the same way like on EVM

**Review by EasterTheBunny — DISMISSED** [review]
LGTM

**Review by EasterTheBunny — DISMISSED** [review]

**Review by jadepark-dev — DISMISSED** [review]
looks good to me

File: pkg/solana/codec/testutils/itemArray1TypeIDL.json [hunk 2]
```diff
@@ -0,0 +1,92 @@
+{
```
Comment (jadepark-dev) [inline pkg/solana/codec/testutils/itemArray1TypeIDL.json:1]: I'm curious if those idls are purely mock data or do we have corresponding program somewhere?

**Review by jadepark-dev — DISMISSED** [review]

File: pkg/solana/codec/testutils/itemArray1TypeIDL.json (hunk 2 above)
Comment (toblich) [reply pkg/solana/codec/testutils/itemArray1TypeIDL.json:1]: Based on where they're used and the names inside of them, it looks to be just for tests

**Review by toblich — DISMISSED** [review]
LGTM

**Review by EasterTheBunny — DISMISSED** [review]

File: pkg/solana/codec/testutils/itemArray1TypeIDL.json (hunk 2 above)
Comment (ilija42) [reply pkg/solana/codec/testutils/itemArray1TypeIDL.json:1]: yeah, just wrote them "manually" no corresponding program

**Review by EasterTheBunny — DISMISSED** [review]

**Review by jadepark-dev — DISMISSED** [review]

**Review by jadepark-dev — APPROVED** [review]

**Review by EasterTheBunny — APPROVED** [review]

**Pull request merged**